File: cs/Iris/ParserUtils/LexParseException/SyntaxError.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Iris.ParserUtils.LexParseException
{
    public class SyntaxError : BaseException
    {
        public Token.TokenValue tok;

        public SyntaxError(int lineNumber, int colNumber, Token.TokenValue expectedTok) : base(lineNumber, colNumber)
        {
            tok = expectedTok;
        }

        public override string ExceptionToString()
        {
            StringBuilder str = new StringBuilder();

            str.Append("syntax error: ");
            str.Append(base.stringTrace());
            str.Append("expected ");
            str.Append(Token.tokenstring(tok));

            return str.ToString();
        }
    }
}

File: java/Iris/src/com/rokuan/iris/exception/SyntaxError.java
package com.rokuan.iris.exception;

import com.rokuan.iris.parser.Token;

public class SyntaxError extends BaseException
{
	public Token.TokenValue tok;

	public SyntaxError(int lineNumber, int colNumber, Token.TokenValue expectedTok)
	{
		super(lineNumber, colNumber);
		tok = expectedTok;
	}

	public String ExceptionToString()
	{
		StringBuilder str = new StringBuilder();

		str.append("syntax error: ");
		str.append(super.stringTrace());
		str.append("expected ");
		str.append(Token.tokenstring(tok));

		return str.toString();
	}
}
